src/Robo/Plugin/Traits/DatabaseDownloadTrait.php

```diff
@@ -49,7 +49,7 @@ public function databaseDownload(string $siteName = 'default')
         ]);
         $objects = $s3->listObjectsV2($this->s3BucketRequestConfig($siteName));
         $objects = iterator_to_array($objects);
-        if (empty($objects)) {
+        if (count($objects) == 0) {
```

Comment:
Again, it looks like we are adding robustness in the calling code checks, rather than to the return of the function. If we don't control the signature of `$s3->listObjectsV2` then this is fine (though I would be tempted then to wrap/extend that method to provide that robustness). Try to get to `if ($objects) {` and force the upstream function to be factored in such a way that guarantees a known type of response.

Reply:
We do not control this. It is an AWS PHP library that we are using. Again, maybe I am being too defensive here.

Reply:
What is the benefit of this change? i.e. what was the linter saying was the issue?

I'm more curious than anything, but I'm also a frequent user of `empty` checks.

Reply:
> We do not control this. It is an AWS PHP library that we are using. Again, maybe I am being too defensive here.

I don't feel you are being defensive :). And if these methods don't guarantee a certain type of return value, then you are doing the right thing.

Reply:
> What is the benefit of this change? i.e. what was the linter saying was the issue?
> 
> I'm more curious than anything, as I'm also a frequent user of empty checks.
> 

@adamzimmermann [phpstan-strict does not like empty()](https://github.com/phpstan/phpstan-strict-rules):

> Disallow empty() - it's a very loose comparison (see [manual](https://php.net/empty)), it's recommended to use more strict one.

Reply:
Ah, see that's what I like about it in some circumstances.